data/plotpaf.rmd

```diff
@@ -166,21 +166,46 @@ pal["Other"] <- "#c0c0c0"
 
 pafq <- pafq_lumped %>% arrange(desc(Qlength), Qname, Qstart, Barcodes)
 
-ggplot(pafq) +
-	geom_rect(aes(
-		xmin = Qstart, xmax = Qend,
-		ymin = Qindex, ymax = Qindex + 1,
-		fill = Tname)) +
-	geom_point(aes(x = Qlength, y = 0.5 + Qindex),
-		data = distinct(pafq, Qlength, Qindex)) +
-	scale_x_continuous(name = "Position", labels = unit_format(unit = "M", scale = 1e-6)) +
-	scale_y_reverse(name = "Chromosome (Query)",
-		breaks = 0.5 + seq_len(nlevels(pafq$Qname)),
-		labels = levels(pafq$Qname)) +
-	scale_fill_manual(name = "Target", values = pal) +
-	theme_bw() +
-	guides(fill = guide_legend(ncol = 1)) +
-	labs(caption = input_paf)
+pafq_ideogram <- pafq %>%
+  arrange(desc(Qlength), Qname, Qstart, Barcodes) %>%
+  group_by(Qname) %>%
+  mutate(Colour = (Qindex + row_number()) %% 2 + 26) %>%
+  ungroup() %>%
+  mutate(Colour = factor(Colour))
+pafq_ideogram$Colour <- as.character(pafq_ideogram$Colour)
+pafq_ideogram$Colour[pafq_ideogram$Colour == "26"] <- "Other1"
+pafq_ideogram$Colour[pafq_ideogram$Colour == "27"] <- "Other2"
+pafq_ideogram$Colour <- as.factor(pafq_ideogram$Colour)
+pal <- setNames(alphabet_colours, pafq_lumped$Tname %>% setdiff("Other") %>% unique())
+pal["Other1"] <- "#808080"
+pal["Other2"] <- "#c0c0c0"
```

Comment:
Try to use `tidyverse` functions rather than `$` and `<-` on individual columns. Once you get in the habit, you'll find the code is more readable and easier to maintain. Use `mutate(Colour = …)` rather than `pafq_ideogram$Colour <-`. Look into the package `forcats` for modifying and reordering levels of a factor variable https://forcats.tidyverse.org. Read Hadley's book R for Data Science https://r4ds.had.co.nz. It's available online, and the lab has a copy. If the lab copy has grown legs and walked away, order a new copy.

Reply:
I used mutate to change the factor statements.